From ece4d22848a7fd0b3b6802e485ff55e80af7a5d8 Mon Sep 17 00:00:00 2001
From: Meir Levi <mlevi4@marvell.com>
Date: Thu, 24 Dec 2020 10:32:46 -0500
Subject: [PATCH 1067/1921] dts: marvell: musdk: add security acceleration
 module UIO node

Patch add Marvell security acceleration module(SAM) UIO
node to device tree for user space SAM application.

Signed-off-by: Meir Levi <mlevi4@marvell.com>
Change-Id: I782643a373f2f689acc56cd69a5dd6d60b76b7c2
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42775
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi  | 5 +++++
 arch/arm64/boot/dts/marvell/armada-cp11x.dtsi | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index f53e4e050c6a..9d23f664f7a1 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -155,6 +155,11 @@
 				status = "disabled";
 			};
 
+			uio-sam {
+				compatible = "marvell,uio-sam";
+				eip_access = <&crypto>;
+			};
+
 			nb_periph_clk: nb-periph-clk@13000 {
 				compatible = "marvell,armada-3700-periph-clock-nb";
 				reg = <0x13000 0x100>;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
index 9dcf16beabf5..db1346ad1be0 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
@@ -490,6 +490,11 @@
 				 <&CP11X_LABEL(clk) 1 17>;
 			dma-coherent;
 		};
+
+		CP11X_LABEL(uio_sam): uio_sam@800000 {
+			compatible = "marvell,uio-sam";
+			eip_access = <&CP11X_LABEL(crypto)>;
+		};
 	};
 
 	CP11X_LABEL(pcie0): pcie@CP11X_PCIE0_BASE {
-- 
2.31.1

